app/views/layouts/_taskbar.html.haml

```diff
@@ -3,8 +3,11 @@
   %div{:style => "float: left;"}
     - defered_toolbar_render('summary_center_tb', controller.send(:restful?) ? "summary_center_restful_tb" : "summary_center_tb")
 
-= render :partial => "layouts/center_buttons"
-= render :partial => "layouts/x_custom_buttons"
+- defered_toolbar_render('center_tb', center_toolbar_filename)
+
+- if fname = custom_toolbar_filename
```

Comment:
@himdel : you are right and actually I had the same thought. In this PR i just moved this login one prartial closer to the controller. I can do that in the next set of patches (separate PR).